ktor-utils/posix/src/io/ktor/util/CryptoNative.kt

```diff
@@ -4,12 +4,37 @@
 
 package io.ktor.util
 
+import kotlinx.cinterop.*
+import platform.Security.*
+
+private const val NONCE_SIZE_IN_BYTES = 16
 
 /**
- * Generates a nonce string. Could block if the system's entropy source is empty
+ * Generates a nonce string 16 characters long. Could block if the system's entropy source is empty
  */
 @InternalAPI
-actual fun generateNonce(): String = error("[generateNonce] is not supported on iOS")
+actual fun generateNonce(): String {
+    return hex(generateNonceByteArray())
+}
+
+private fun generateNonceByteArray(): ByteArray {
+    memScoped {
+        val array = allocArray<ByteVarOf<Byte>>(NONCE_SIZE_IN_BYTES)
+        val status = SecRandomCopyBytes(
+            kSecRandomDefault,
+            NONCE_SIZE_IN_BYTES.toULong(),
+            array
+        )
+        if (status == errSecSuccess) {
+            val result = ByteArray(NONCE_SIZE_IN_BYTES)
+            for (i in 0 until NONCE_SIZE_IN_BYTES) {
+                result[i] = array[i]
+            }
+            return result
+        }
+        error("Nonce could not be generated")
```

Comment:
Can we have some more debug info here? It is slightly strange that the documentation doesn't state what kind of errors may happen and what do they mean

Reply:
I checked the web a bit and found the following: 
[https://stackoverflow.com/questions/27011825/random-256bit-key-using-secrandomcopybytes-in-ios](https://stackoverflow.com/questions/27011825/random-256bit-key-using-secrandomcopybytes-in-ios
)

In general I don't see this failing. But to have more feedback I added the result code to the error message for easier debugging.